specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,707 @@
+          "type": "string",
+          "example": "2020-09-08T15:55:41Z"
+        },
+        "roomOpen": {
+          "description": "Flag to specify if the room is to be an open room or closed room.",
+          "type": "boolean",
+          "example": false
+        },
+        "participants": {
+          "description": "(Optional) Collection of identities invited to the room.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoomParticipant"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "RoomParticipant": {
+      "description": "A participant of the room.",
+      "required": [
+        "communicationIdentifier"
+      ],
+      "type": "object",
+      "properties": {
+        "communicationIdentifier": {
+          "$ref": "#/definitions/CommunicationIdentifierModel"
+        },
+        "role": {
+          "description": "Role Name.",
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationIdentifierModel": {
+      "description": "Identifies a participant in Azure Communication services. A participant is, for example, an Azure communication user. This model must be interpreted as a union: Apart from rawId, at most one further property may be set.",
+      "type": "object",
+      "properties": {
+        "rawId": {
+          "description": "Raw Id of the identifier. Optional in requests, required in responses.",
+          "type": "string"
+        },
+        "communicationUser": {
+          "$ref": "#/definitions/CommunicationUserIdentifierModel"
+        }
+      }
+    },
+    "CommunicationUserIdentifierModel": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationUserIdentifierModel"
+    },
+    "RoomModel": {
+      "description": "The meeting room.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique identifier of a room. This id is server generated.",
+          "type": "string",
+          "example": "999126454"
+        },
+        "createdDateTime": {
+          "format": "date-time",
+          "description": "The timestamp when the room was created at the server. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2021-09-06T05:55:41Z"
+        },
+        "validFrom": {
+          "format": "date-time",
+          "description": "The timestamp from when the room is open for joining. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2021-09-07T07:55:41Z"
+        },
+        "validUntil": {
+          "format": "date-time",
+          "description": "The timestamp from when the room can no longer be joined. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2021-09-08T15:55:41Z"
+        },
+        "roomOpen": {
```

Comment:
As a general principle, avoid boolean flags in API models.
It's worth thinking about whether this should be an extensible enum instead `guestPolicy: 'allow' | 'disallow'`.

The open/close room concept wasn't immediately obvious to me. If I understand correctly, in an open room, anybody with the `roomId` can join an ongoing call in the room (presumably only if there is already an ongoing call?). 

I can think of other interpretations of open/close for a room: whether you can add more participants once the room is created or not.

I can also think of a couple more values for the extensible `guestPolicy` enum: `allowWithPrompt` -- an organizer has to manually allow the guest in a lobby; `allowStartingCall` -- a guest can even start a call in the room as long as they have the `roomId`. Those values may not make sense right now, but an extensible enum would leave the door open for them.

Reply:
Addressed: changed from string to extensible enum.
roomOpen is replaced by roomJoinPolicy which is an extensible enum with values ["InviteOnly", "CommunicationServiceUsers"],